src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,920 @@
+                    queryExpressionProcessingInfo);
+                return new DocumentUpdateInfo(forEachStatement, statements);
+            }
+
+            private bool TryConvertIfInReturnStatement(
+                ReturnStatementSyntax returnStatement,
+                QueryExpressionProcessingInfo queryExpressionProcessingInfo,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                // The conversion requires yield return which cannot be added to lambdas and anonymous method declarations.
+                if (IsWithinImmediateLambdaOrAnonymousMethod(returnStatement))
+                {
+                    documentUpdateInfo = default;
+                    return false;
+                }
+
+                var memberDeclarationNode = FindParentMemberDeclarationNode(returnStatement);
+                if (memberDeclarationNode == null)
+                {
+                    throw new ArgumentException("Must bail out on previous steps");
+                }
+
+                var declaredSymbol = _semanticModel.GetDeclaredSymbol(memberDeclarationNode);
+                ITypeSymbol typeSymbol;
+                switch (declaredSymbol.Kind)
+                {
+                    case SymbolKind.Local:
+                        typeSymbol = ((ILocalSymbol)declaredSymbol).Type;
+                        break;
+                    case SymbolKind.Method:
+                        typeSymbol = ((IMethodSymbol)declaredSymbol).ReturnType;
+                        break;
+                    case SymbolKind.Property:
+                        typeSymbol = ((IPropertySymbol)declaredSymbol).Type;
+                        break;
+                    default:
+                        throw new ArgumentException(declaredSymbol.Kind.ToString());
+                }
+
+                if (typeSymbol.OriginalDefinition?.SpecialType != SpecialType.System_Collections_Generic_IEnumerable_T)
+                {
+                    documentUpdateInfo = default;
+                    return false;
+                }
+
+                // before: return from a in b select a;
+                // after: 
+                // foreach(var a in b)
+                // {
+                //      yield return a;
+                // }
+                var statements = GenerateStatements((ExpressionSyntax expression)
+                    => SyntaxFactory.YieldStatement(SyntaxKind.YieldReturnStatement, expression), queryExpressionProcessingInfo);
+
+
+                // if there are more than one return in the method, add yield break;
+                if (memberDeclarationNode.DescendantNodes().OfType<ReturnStatementSyntax>().Count() != 1)
+                {
+                    var yieldBreakStatement = SyntaxFactory.YieldStatement(SyntaxKind.YieldBreakStatement);
+                    documentUpdateInfo = new DocumentUpdateInfo(returnStatement, statements.Concat(new[] { yieldBreakStatement }));
```

Comment:
>new DocumentUpdateInfo(returnStatement, statements.Concat(new[] { yieldBreakStatement })); [](start = 41, length = 90)

Everywhere where we replace a single statement with a bunch of statements, we need to make sure that we don't need to wrap them with a block. For example, when the statement we replace is an embedded statement in ```if```, ```for```, etc.